chain/client/src/sync.rs

```diff
@@ -823,13 +823,24 @@ impl StateSync {
                             status: ShardSyncStatus::StateSplit,
                         }
                     } else {
+                        *shard_sync_download = ShardSyncDownload {
+                            downloads: vec![],
+                            status: ShardSyncStatus::BlocksCatchingUp,
+                        };
```

Comment:
Ideally we should separate the two since catching up block is functionally not part of state sync. Mixing them together also makes state sync even more complex and harder to reason about

Reply:
We need a way to skip through state downloading part of sync without doing anything, this seems the easiest way to do that

Reply:
> We need a way to skip through state downloading part of sync without doing anything

What is the reason for that?

Reply:
Yeah I'm not sure how the `BlocksCatchingUp` logic here actually does anything. I don't see it used anywhere else (maybe I am missing something). So it seems to just add a stage after either `StateSplit` or `StateDownloadComplete`, in which nothing happens?

Anyway, I agree with Bowen that this logic here is quite confusing.

Reply:
Actually our current implementation is buggy if there are more than 2 shards. If we need different number of sync steps for different shards (due to download speed, for example), at some point one for the shards would get into StateDownloadComplete state earlier then others, and on next step it would execute code that is executed under this state again. Although I agree that maybe we should name new enum variant some other way to indicate we are just done with this shard and nothing need to be done here

Reply:
Ah that's a good catch! 

Maybe we should name the new stage to `StateSyncDone`? That signifies that it's the end of state sync for this shard.

Reply:
I like this name